compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -62,6 +62,29 @@ pub(crate) struct GeneratedAcir {
     /// Name for the corresponding entry point represented by this Acir-gen output.
     /// Only used for debugging and benchmarking purposes
     pub(crate) name: String,
+
+    /// Maps the opcode index to a Brillig std library function call.
+    /// As to avoid passing the ACIR gen shared context into each individual ACIR
+    /// we can instead keep this map and resolve the Brillig calls at the end of code generation.
+    pub(crate) brillig_stdlib_func_locations: BTreeMap<OpcodeLocation, BrilligStdlibFunc>,
```

Comment:
I did consider potentially unifying this approach with the existing `BrilligCall` resolution, but didn't want to add too much to this PR. We would then also have an additional map of `BTreeMap<OpcodeLocation, FunctionId>` for normal Brillig calls on the `GeneratedAcir`, and would resolve the function pointers after ACIR gen as this PR does for the Brillig stdlib. We could get rid of the `SharedContext` lifetime on the `Context` struct. I still felt this refactor would be best left separated from the Brillig stdlib.